### PR TITLE
[FEATURE] DevBox incl. Breakpoints & App Context

### PR DESCRIPTION
**Pull request by vkemeter**
- added some patterns to the gitignore
- added bootstrap scss source files to the gitignore
- added a context viewhelper which simply returns the application context
- added missing page = PAGE in the page setup
- added compiled css file
- added some scss source files (only custom stuff) where i included the
- bootstrap scss files. recompiling this files would end in an error ;)

at the end there is a dev box on the bottom left corner which shows you
the breakpoint and the application context.

sometimes it is freakn helpfull ;)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,3 +17,17 @@ nbproject
...
+*.tmrproj
+*.iml
+.Trashes
+Thumbs.db
+ehthumbs.db
+Desktop.ini
+@eaDir
+
+# bootstrap scss files
+app/web/typo3conf/ext/theme/Resources/Private/Scss/bootstrap
```
Comment (jousch) [inline .gitignore]: Generally .gitignore in root of the distribution should not include application relevant stuff

File: .gitignore [hunk 2]
```diff
@@ -17,3 +17,17 @@ nbproject
 *.zip
 *.bak
 /backup/*
+*.bak
+*~
+.htaccess
```
Comment (jousch) [inline .gitignore]: What are the benefits of not commiting .htaccess?

File: app/.gitignore [hunk 3]
```diff
@@ -28,6 +28,8 @@
 /web/fileadmin/_processed_/**/*
 /web/uploads/**/*
 
+# remove the cssmapping files
+*.map
```
Comment (jousch) [inline app/.gitignore]: Would you think it makes sense to ignore them globally, or just within the dist folder (something like `app/web/typo3conf/ext/theme/Resources/Public/SourceMaps/**/*`)

File: app/web/typo3conf/ext/theme/Classes/ViewHelpers/ScssBreakpointViewHelper.php [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+use \TYPO3\CMS\Extbase\Utility\DebuggerUtility;
+use TYPO3\CMS\Core\Utility\ExtensionManagementUtility;
+use TYPO3\CMS\Core\Utility\GeneralUtility;
+use TYPO3\CMS\Core\Page\PageRenderer;
+use function GuzzleHttp\json_encode;
+
+/**
+ * parses a given scss file and extracts the breakpoint variables
+ */
+class ScssBreakpointViewHelper extends AbstractViewHelper {
```
Comment (jousch) [inline app/web/typo3conf/ext/theme/Classes/ViewHelpers/ScssBreakpointViewHelper.php]: I like the approach of having all bs4 breakpoints dynamically imported thingi. But after we talked about to not deploy scss to production, what do you think?

File: app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/Development.html [hunk 5]
```diff
@@ -0,0 +1,7 @@
+{namespace s7=Sup7even\Theme\ViewHelpers}
+
+<f:if condition="{s7:context()} != 'Production'">
```
Comment (jousch) [inline app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/Development.html]: This condition fails on common application context `Production/Live`

File: app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/InlineJS.html [hunk 6]
```diff
@@ -0,0 +1,10 @@
+{namespace s7=Sup7even\Theme\ViewHelpers}
+
+<f:comment>
+	@TODO: change the inline js with a addjs method in the viewhelper. that
+	could be a bit more nice ;) especially with the concatenation and compression
+	typo3 stuff shizzle.
```
Comment (jousch) [inline app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/InlineJS.html]: 👍

File: app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/Development.html (hunk 5 above)
Comment (jousch) [inline app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/Development.html]: Maybe it's better to set the state on/off for this feature via fluid setting in TypoScript?

File: app/web/typo3conf/ext/theme/Resources/Private/Scss/page/development.scss [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+			@include media-breakpoint-only(md) {
+				content: 'MD';
+			}
+
+			@include media-breakpoint-only(lg) {
+				content: 'LG';
+			}
+
+			@include media-breakpoint-only(xl) {
```
Comment (jousch) [inline app/web/typo3conf/ext/theme/Resources/Private/Scss/page/development.scss]: What about generating scss/css also automatically inline within the partial?

File: .gitignore (hunk 1 above)
Comment (vkemeter) [inline .gitignore]: we already discussed that ;) feel free to kick it. if we use stuff like npm and more that is surely not needed. ;)

File: app/web/typo3conf/ext/theme/Classes/ViewHelpers/ScssBreakpointViewHelper.php (hunk 4 above)
Comment (vkemeter) [inline app/web/typo3conf/ext/theme/Classes/ViewHelpers/ScssBreakpointViewHelper.php]: yeah. thats right. if we dont deploy scss files, we may write a little config file (json or whatevertype) for that purpose, maybe we can create that file dynamically in that viewhelper. if that file does not exist, create a "breakpoints.json" file for production mode eg.

File: app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/Development.html (hunk 5 above)
Comment (vkemeter) [inline app/web/typo3conf/ext/theme/Resources/Private/Partials/Page/Development.html]: that is just an example. it is on your own, how you want to apply that small div layer in your template. the main logic happens in the scss/css file with after/before things.

File: app/web/typo3conf/ext/theme/Resources/Private/Scss/page/development.scss (hunk 7 above)
Comment (vkemeter) [inline app/web/typo3conf/ext/theme/Resources/Private/Scss/page/development.scss]: dont like the idea of inlining stuff in vh. dont know why.

File: app/.gitignore (hunk 3 above)
Comment (vkemeter) [inline app/.gitignore]: i guess it makes sense to ignore them globally, because with every compiling, the map files are regenerated.

**Pull request closed without merging**